tests/test_e2e_80_pathfinder.py

```diff
@@ -0,0 +1,160 @@
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+"source": "00:00:00:00:00:00:00:01:3",
+"destination": "00:00:00:00:00:00:00:06:3",
+"undesired_links": undesired_link,
+"spf_attribute": "hop",
+"spf_max_paths": 1,
+"parameter": "hop"
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert data['paths'][0]['cost'] == expected, f'Shortest path expected {expected}: {data}'
+
+    @pytest.mark.parametrize("attribute, expected_cost", 
+                            [("hop", 1), ("delay", 20), ("priority", 15)])
+    def test_20_spf_attribute_and_max_path_cost(self, attribute, expected_cost):
+        """Tests hop, delay, and priority with path cost"""
+        links_metadata = self.add_topology_metadata()
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+"source": "00:00:00:00:00:00:00:01:3",
+"destination": "00:00:00:00:00:00:00:06:3",
+"spf_attribute": attribute,
+"spf_max_path_cost": expected_cost,
+"parameter": attribute
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert data['paths'][0]['cost'] == expected_cost, f'Path cost larger than {expected_cost}: {data}'
+
+    def test_30_mandatory_metrics(self):
+        """The returned path should be of ownership blue"""
+        links_metadata = self.add_topology_metadata()
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+"source": "00:00:00:00:00:00:00:01:3",
+"destination": "00:00:00:00:00:00:00:06:3",
+"spf_attribute": "hop",
+"mandatory_metrics": {
+"ownership": "blue"
+}
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert data['paths'][0]['metrics']['ownership'] == "blue", f'Path ownership not blue: {data}'
+
+    @pytest.mark.parametrize("ownership, expected_cost", 
+                            [("blue", 2), ("red", 0)])
+    def test_40_flexible_metrics_and_hits(self, ownership, expected_cost):
+        """First test has all required metrics and will return paths, second test has no paths since no link meets the required flexible metrics"""
+        links_metadata = self.add_topology_metadata()
+        api_url = KYTOS_API + '/pathfinder/v3/'
+        post_body = {
+  "source": "00:00:00:00:00:00:00:01:3",
+  "destination": "00:00:00:00:00:00:00:06:3",
+  "spf_attribute": "hop",
+  "flexible_metrics": {
+  "delay": 10,
+  "ownership": ownership
+},
+  "minimum_flexible_hits": 2
+}
+
+        response = requests.post(api_url, json=post_body)
+        assert response.status_code == 200, response.text
+        data = response.json()
+        assert len(data['paths']) == expected_cost, f'Path cost larger than {expected_cost}: {data}'
```

Comment:
Same as prior comment assert the vertices of the minimum expected path too. Also,  if you could give a more descriptive name to `expected_cost`.  `expected_cost` is behaving like like `expected_n_paths` or something similar here.